### PR TITLE
Fix broken A/D Controller metrics test

### LINKED ISSUE #126505: Failure cluster [47ae877c...] e2e-ci-kubernetes-e2e-al2023-aws-serial-canary issues
##### Error text:
```
[FAILED] Key "attachdetach_controller_forced_detaches" not found in A/D Controller metrics
In [It] at: k8s.io/kubernetes/test/e2e/storage/volume_metrics.go:436 @ 07/18/24 00:34:39.158

There were additional failures detected after the initial failure. These are visible in the timeline

```
#### Recent failures:
[7/31/2024, 8:06:51 PM e2e-ci-kubernetes-e2e-al2023-aws-serial-canary](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/e2e-ci-kubernetes-e2e-al2023-aws-serial-canary/1818800431434829824)
[7/31/2024, 2:06:51 PM e2e-ci-kubernetes-e2e-al2023-aws-serial-canary](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/e2e-ci-kubernetes-e2e-al2023-aws-serial-canary/1818709833696677888)
[7/31/2024, 10:39:42 AM e2e-ci-kubernetes-e2e-al2023-aws-serial-canary](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/e2e-ci-kubernetes-e2e-al2023-aws-serial-canary/1818657698065944576)
[7/23/2024, 2:07:11 AM e2e-ci-kubernetes-e2e-al2023-aws-serial-canary](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/e2e-ci-kubernetes-e2e-al2023-aws-serial-canary/1815629619794022400)
[7/22/2024, 8:07:10 PM e2e-ci-kubernetes-e2e-al2023-aws-serial-canary](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/e2e-ci-kubernetes-e2e-al2023-aws-serial-canary/1815539020898242560)

/kind failing-test

/sig storage

### PR DESCRIPTION
**Pull request by AndrewSirenko**
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:

Test `[sig-storage] [Serial] Volume metrics PVC should create metrics for total number of volumes in A/D Controller` has been failing due to attach due to: 

```
[FAILED] Key "attachdetach_controller_forced_detaches" not found in A/D Controller metrics
```

Seems that metric is not being emitted during test, likely because force detach is not being exercised within it  ->  metric never initialized.

No required pre-submits run this test, and most periodicals skip it because this test requires certain providers or a default storage class. However,`e2e-ci-kubernetes-e2e-cos-gce-serial-canary` & `e2e-ci-kubernetes-e2e-al2023-aws-serial-canary` have been failing for a while. 

Has this test really been broken for years? @torredil showed yes by looking all the way back to 2022... 

#### Which issue(s) this PR fixes:

Fixes #126505

#### Special notes for your reviewer:

Tests pass locally on kOps cluster. 

These volume metrics tests are not currently run on any presubmit... See #126345 

TLDR no pre-submit is serial && is one of non-skipped provider && and has defaultSC set up...

```
❯ ginkgo run --focus 'should create metrics for total number of volumes in A/D Controller' ./test/e2e --fast-fail -- --provider aws
  I0801 21:59:30.076478 1975610 e2e.go:109] Starting e2e run "821bb17b-343c-43aa-81b1-9cc13c3b70d6" on Ginkgo node 1
Running Suite: Kubernetes e2e suite - /home/andsirey/workplace/kubernetes/test/e2e
==================================================================================
Random Seed: 1722549557 - will randomize all specs

Will run 2 of 6603 specs
...

Ran 2 of 6603 Specs in 44.872 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 6601 Skipped
PASS

Ginkgo ran 1 suite in 57.150179803s
Test Suite Passed
```

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by AndrewSirenko** [comment]
/CC @dims

Should I add a "check forced detach metrics" test in this PR? Or is just opening an issue fine. 

And should we wait until some pre-submit can test this, or do we want to check if this will fix flakes.

**Comment by dims** [comment]
/assign @xing-yang @msau42 

@AndrewSirenko please do reach out to the sig-storage leads :)

**Comment by msau42** [comment]
I think we would need to test force detach using csi-mock driver where we could inject errors into the driver. So I am ok with removing this. @xing-yang not sure if the ungraceful shutdown tests can cover this?

**Comment by xing-yang** [comment]
> I think we would need to test force detach using csi-mock driver where we could inject errors into the driver. So I am ok with removing this. @xing-yang not sure if the ungraceful shutdown tests can cover this?

The non-graceful node shutdown e2e test was not merged due to concerns that it depends on specific cloud providers:
https://github.com/kubernetes/kubernetes/pull/118848

**Comment by carlory** [comment]
> I think we would need to test force detach using csi-mock driver where we could inject errors into the driver. So I am ok with removing this. @xing-yang not sure if the ungraceful shutdown tests can cover this?

@AndrewSirenko can you convert it into an issue? this PR will be merged after 1.31 is released because it has been LGTMed and approved.
